osu.Game/Screens/Select/BeatmapDetails.cs

```diff
@@ -50,7 +50,7 @@ public BeatmapInfo Beatmap
 
             set
             {
-                if (beatmap == value) return;
+                if (beatmap == value || value.OnlineBeatmapSetID < 1) return;

```

Comment:
They do if they are uploaded, but we also need to support local beatmaps. Also, the beatmap set ID is unreliably. It's not always correct. So this is probably not a solution that'll work in the general case.